designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/NodesApiEndpoints.scala

```diff
@@ -587,6 +652,78 @@ object NodesApiEndpoints {
           )
       )
 
+    val sourceCompilationExample: EndpointOutput.OneOfVariant[SourceCompilation] =
+      oneOfVariantFromMatchType(
+        BadRequest,
+        plainBody[SourceCompilation]
+          .example(
+            Example.of(
+              summary = Some("Source compilation error"),
+              value = SourceCompilation("sourceId", List("Invalid source configuration", "Missing required parameter"))
+            )
+          )
+      )
+
+    val unsupportedSourcePreviewExample: EndpointOutput.OneOfVariant[UnsupportedSourcePreview] =
+      oneOfVariantFromMatchType(
+        NotFound,
+        plainBody[UnsupportedSourcePreview]
+          .example(
+            Example.of(
+              summary = Some("Source preview not supported"),
+              value = UnsupportedSourcePreview("sourceId")
+            )
+          )
+      )
+
+    val noDataGeneratedExample: EndpointOutput.OneOfVariant[NoDataGenerated.type] =
+      oneOfVariantFromMatchType(
+        NotFound,
+        plainBody[NoDataGenerated.type]
+          .example(
+            Example.of(
+              summary = Some("No test data generated"),
+              value = NoDataGenerated
+            )
+          )
+      )
+
+    val noSourcesWithTestDataGenerationExample: EndpointOutput.OneOfVariant[NoSourcesWithTestDataGeneration.type] =
```

Comment:
This one won't happen because it is a valid error for test generation based on scenario, not a node. Maybe we could another sealed trait level below `ScenarioTestError` - `SourceTestError`. WDYT?

Reply:
I tried to make class hierarchy where ScenarioError extends SourceError, but it didn't really work too well with Eithers. This is why I duplicated classes a little bit and I made it a flat structure for both cases. I don't have any better ideas for now. WDYT?